File: selenium-agent-mcp-starter/pages/cart.page.ts
import { By } from 'selenium-webdriver';
import { BasePage } from './base.page';

export class CartPage extends BasePage {
  private title = By.css('.title');
  private cartItems = By.css('.cart_item');
  private checkoutButton = By.id('checkout');
  private continueShoppingButton = By.id('continue-shopping');
  private removeButton = By.css('[data-test^="remove"]');

  async getTitleText(): Promise<string> {
    return this.getText(this.title);
  }

  async getCartItemCount(): Promise<number> {
    const items = await this.getElements(this.cartItems);
    return items.length;
  }

  async proceedToCheckout(): Promise<void> {
    await this.jsClick(this.checkoutButton);
    await this.waitForUrlContains('checkout-step-one.html');
  }

  async continueShopping(): Promise<void> {
    await this.jsClick(this.continueShoppingButton);
    await this.waitForUrlContains('inventory.html');
  }

  async removeFirstItem(): Promise<void> {
    await this.jsClick(this.removeButton);
  }
}

File: selenium-agent-mcp-starter-python/src/pages/cart_page.py
from selenium.webdriver.common.by import By
from src.pages.base_page import BasePage


class CartPage(BasePage):
    _title = (By.CSS_SELECTOR, ".title")
    _cart_items = (By.CSS_SELECTOR, ".cart_item, [data-test='inventory-item']")
    _checkout_button = (By.ID, "checkout")
    _continue_shopping_button = (By.ID, "continue-shopping")
    _remove_backpack_button = (
        By.XPATH,
        "//div[contains(@class,'cart_item')][.//div[contains(@class,'inventory_item_name') and normalize-space()='Sauce Labs Backpack']]//button[contains(@id,'remove') or normalize-space()='Remove']",
    )

    def is_loaded(self) -> bool:
        return "cart.html" in self.current_url() and self.is_visible(self._checkout_button)

    def get_item_count(self) -> int:
        return len(self.find_elements(self._cart_items))

    def remove_backpack(self):
        self.js_click(self._remove_backpack_button)

    def continue_shopping(self):
        self.js_click(self._continue_shopping_button)
        self.wait_for_url_contains("inventory.html")

    def checkout(self):
        self.js_click(self._checkout_button)
        self.wait_for_url_contains("checkout-step-one.html")
